libraries/AP_Beacon/AP_Beacon.cpp

```diff
@@ -235,15 +235,12 @@ bool AP_Beacon::update_boundary_points()
         return false;
     }
 
-    // delete to recreate boundary
-    if (_boundary != nullptr) {
-        delete _boundary;
-        _boundary = nullptr;
+    // create boundary (5 points with a closing point) at once
+    if (_boundary == nullptr) {
```

Comment:
Define five points of becon in the HEAP area like acquiring the area of WP. I think that it should be obtained from that area.
EX:
/// load polygon points stored in eeprom into boundary array and perform validation
bool AC_Fence::load_polygon_from_eeprom(bool force_reload)
{
    // exit immediately if already loaded
    if (_boundary_loaded && !force_reload) {
        return true;
    }

    // check if we need to create array
    if (!_boundary_create_attempted) {
        _boundary = (Vector2f *)_poly_loader.create_point_array(sizeof(Vector2f));
        _boundary_create_attempted = true;
    }

    // exit if we could not allocate RAM for the boundary
    if (_boundary == nullptr) {
        return false;
    }